src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -439,6 +439,35 @@ class IndexLogEntryTest extends SparkFunSuite with SQLHelper with BeforeAndAfter
     FileUtils.deleteDirectory(tempDir.toFile)
   }
 
+  test("Directory.merge Test: test if merge works as expected.") {
+    // File Structure
+    // testDir/a/f1
+    // testDir/a/f2
+    // testDir/a/b/f3
+    // testDir/a/b/f4
+    // Expected Result: Merged Directory structure when merging Directory object of dir a with
+    // that of b.
+    val tempDir = Files.createDirectories(Paths.get(testDir + "/temp"))
+    val a = Files.createDirectories(Paths.get(tempDir + "/a"))
+    val b = Files.createDirectories(Paths.get(a + "/b"))
+    val f1 = Files.createFile(Paths.get(a + "/f1"))
+    val f2 = Files.createFile(Paths.get(a + "/f2"))
+    val f3 = Files.createFile(Paths.get(b + "/f3"))
+    val f4 = Files.createFile(Paths.get(b + "/f4"))
```

Comment:
I think we need to handle the case where the same sub dir has files?
For example, `bDirectory` can have `testDir/a/f5`.

Reply:
thanks @imback82 , added